### PR TITLE
Update golang version to 1.22.5

### PR DESCRIPTION
**Pull request by msftpgayam**

### CODE REVIEW

**Review by madhanrm — DISMISSED** [review]

File: pkg/validations/proxy_validation_test.go [hunk 1]
```diff
@@ -38,11 +39,8 @@ func Test_TestProxyUrlConnection(t *testing.T) {
 	parsedUrl, _ := ValidateProxyURL("http://w3proxy.netscape.com:3128")
 	// Invalid hostname
 	err = TestProxyUrlConnection(parsedUrl, caCertString, "")
-	expectedResult := "Get \"https://mcr.microsoft.com\": proxyconnect tcp: dial tcp: lookup w3proxy.netscape.com: no such host: Invalid Input"
-
-	if err.Error() != expectedResult {
-		t.Fatalf("Test_TestProxyUrlConnection test case failed. Expected error %s but got %s", expectedResult, err.Error())
-	}
+	expectedResult := "Get \"https://mcr.microsoft.com\": proxyconnect tcp: dial tcp: lookup w3proxy.netscape.com"
+	assert.ErrorContains(t, err, expectedResult)
```
Comment (msftpgayam) [inline pkg/validations/proxy_validation_test.go:43]: let me fix this unittest

**Review by nikohobart — APPROVED** [review]

**Review by ganeshamohana — APPROVED** [review]

**Pull request merged**